### PR TITLE
Adds  Compendium Client module

### PR DESCRIPTION
**Pull request by juanpedromoreno**
This PR adds the same client we had in compendium here in the plugin, as it needs to be valid for SBT usage. Additionally, I'm adding some SBT improvements in terms of settings and modules organization.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ scala:
 
 script:
 - sbt ++$TRAVIS_SCALA_VERSION orgScriptCI
-- sbt ++$TRAVIS_SCALA_VERSION scripted
+- sbt ++$TRAVIS_SCALA_VERSION publishLocal scripted
```
Comment (JesusMtnez) [inline .travis.yml:12]: Why adding `publishLocal` here?

Comment (juanpedromoreno) [reply .travis.yml:12]: By default, scripted doesn't make the publishLocal for the dependent libraries (client)

**Review by ememmr — APPROVED** [review]

**Review by JesusMtnez — APPROVED** [review]

**Comment by JesusMtnez** [comment]
@mrtmmr @juanpedromoreno are we going to completely drop the use of `compendium-client` from sbt-compendium? Or is this just to unlock current development?

**Comment by juanpedromoreno** [comment]
sbt-compendium will have its own client from now on. I think binary clients will cause binary issues down the road. I prefer to have them decoupled.

**Comment by juanpedromoreno** [comment]
The next step would be to remove the dependency with compendium-common.

**Comment by JesusMtnez** [comment]
IMHO, I think it is not as problematic as you suggest having the `client` lib in `compendium` and using it in this plugin. I think we could keep the compatibility and maintain versions using `series/0.2` `series/0.3` schema for versioning in both components and keep maintenance for versions as much as we need keeping the binary compatibility. 

I think we should bring this topic into consideration in our next meet, to agree on which strategy we want to apply. What do you think?

CC @marfacio @pepegar @pirita

**Comment by juanpedromoreno** [comment]
That sounds great to me :) . Just one additional note, the entire ecosystem we are building (mu-compendium-skeuomorph) is aiming code generation instead of binary dependencies. In the medium term, what we will have in sbt-compendium, will be a swagger spec and the client should be auto-generated ;)

**Pull request merged**